src/crud.js

```diff
@@ -0,0 +1,110 @@
+import { clearing, clear } from './clear.js';
+
+const taskListDisplay = document.querySelector('#data');
+const inputValue = document.querySelector('.new-task');
+const addButton = document.querySelector('.add-to-list');
+const deleteBtn = document.querySelector('.delete');
+
+let taskList = [];
+
+const display = () => {
+  taskListDisplay.innerHTML = '';
+  taskList = JSON.parse(localStorage.getItem('localItem')) || [];
+  taskList.forEach((element) => {
+    const task = document.createElement('div');
+    const comp = element.completed ? 'checked' : '';
+    task.classList.add('task');
+    task.setAttribute('draggable', 'true');
+    task.innerHTML = `
+    <input type="checkbox" class="check" ${comp} id="check" data-set="${element.index}">
+    <input class="edit" type="text" value="${element.description}">
+    <div class="can">
+    <i id="${element.index}" class="fa-solid fa-trash-can"></i>
+    </div>
+    `;
+    taskListDisplay.appendChild(task);
+
+    task.addEventListener('dragstart', (e) => {
+      const index = Array.from(taskListDisplay.children).indexOf(e.target);
+      e.dataTransfer.setData('text/plain', index);
+    });
+
+    const taskList2 = task.children[1];
+    taskList2.addEventListener('change', () => {
+      const listingData = document.querySelector('#data');
+      const arrayList = Array.from(listingData.children);
+      const index = arrayList.indexOf(task);
+      const taskLocal = JSON.parse(localStorage.getItem('localItem'));
+      taskLocal[index].description = taskList2.value;
+      localStorage.setItem('localItem', JSON.stringify(taskLocal));
+    });
+  });
+};
+
+display();
+
+addButton.addEventListener('click', (e) => {
+  e.preventDefault();
+  if (inputValue.value === '') return;
+  taskList = JSON.parse(localStorage.getItem('localItem')) || [];
+
+  const object = {
+    description: inputValue.value,
+    index: taskList.length + 1,
+    completed: false,
+  };
+  taskList.push(object);
+  localStorage.setItem('localItem', JSON.stringify(taskList));
+
+  inputValue.value = '';
+  display();
+});
+
+const remove = (index) => {
```

Comment:
- [ ] In crud.js: I would suggest you name your variables in a way that they reveal the intention behind it. You could rename **const remove** to **const removeTodo**